compiler/compiler.cc

```diff
@@ -206,6 +206,7 @@ CompilerTargetPlatformToCompilerMSLTargetPlatform(
       return spirv_cross::CompilerMSL::Options::Platform::iOS;
     case Compiler::TargetPlatform::kMacOS:
     // Unknown should not happen due to prior validation.
+    case Compiler::TargetPlatform::kSkSLTranspiler:
```

Comment:
@dnfield @chinmaygarde did you have any suggestions for a better name and/or command-line flag for this?

Reply:
Perhaps something like `DartSPIRV` or `FlutterSPIRV`?

We didn't really come up with any special name for what flutter.dev/go/shaders uses. 

Alternatively, since this just outputs spirv, we could just call it spirv and document that it's intentionally limited to a subset of spirv and will be deprecated in the near future.

Reply:
It's a subset of spirv that the Dart SPIRV to SkSL compiler can understand.

Reply:
I went with `TargetPlatfrom::FlutterSPIRV` and `--flutter-spirv`.